File: 59.spiral-matrix-ii.py
class Solution:
    def generateMatrix(self, n):
        """
        :type n: int
        :rtype: List[List[int]]
        """
        ans = [[0 for _ in range(n)] for _ in range(n)]
        num = 1
        for i in range(0, n // 2):
            for j in range(i, n - i):
                ans[i][j] = num
                num += 1
            for j in range(i + 1, n - 1 - i):
                ans[j][n - 1 - i] = num
                num += 1
            for j in range(n - 1 - i, i - 1, -1):
                ans[n - 1 - i][j] = num
                num+= 1
            for j in range(n - 1 - i - 1, i, -1):
                ans[j][i] = num
                num += 1
        if n % 2:
            ans[n // 2][n // 2] = num
        return ans

File: 59.spiral-matrix-ii.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>> ans(n, vector<int>(n));
        int num = 1;
        for (int i = 0; i < n / 2; i++) {
            for (int j = i; j < n - i; j++) {
                ans[i][j] = num++;
            }
            for (int j = i + 1; j < n - 1 - i; j++) {
                ans[j][n - 1 - i] = num++;
            }
            for (int j = n - 1 - i; j >= i; j--) {
                ans[n - 1 - i][j] = num++;
            }
            for (int j = n - 1 - i - 1; j > i; j--) {
                ans[j][i] = num++;
            }
        }
        if (n % 2) {
            ans[n / 2][n / 2] = num;
        }
        return ans;
    }
};